lambdas/pkgevents/index.py

```diff
@@ -0,0 +1,102 @@
+import itertools
+import json
+import re
+
+import boto3
+
+from t4_lambda_shared.utils import get_quilt_logger
+
+EXPECTED_POINTER_SIZE = 64
+
+event_bridge = boto3.client('events')
+s3 = boto3.client('s3')
+logger = get_quilt_logger()
+
+
+class PutEventsException(Exception):
+    pass
+
+
+class EventsQueue:
+    MAX_SIZE = 10
+
+    def __init__(self):
+        self._events = []
+
+    def append(self, event):
+        self._events.append(event)
+        if len(self) >= self.MAX_SIZE:
+            self._flush()
+
+    def _flush(self):
+        events = self._events
+        self._events = []
+        resp = event_bridge.put_events(Entries=events)
+        if resp['FailedEntryCount']:
+            raise PutEventsException(resp)
+
+    def flush(self):
+        if self:
+            self._flush()
+
+    def __len__(self):
+        return len(self._events)
+
+    def __bool__(self):
+        return bool(self._events)
+
+
+PKG_POINTER_REGEX = re.compile(r'\.quilt/named_packages/([\w-]+/[\w-]+)/(.{10})')
```

Comment:
I think 10 chars is way to specific. It doesn't handle pointers, for instance (which I realize we don't respond to, but we will). If you really only care about numeric timestamps then `\d{10}` but again too specific? I would take something more like `(.+)$`.